### PR TITLE
Update Spatterlight.app to v0.5.12b

### PR DESCRIPTION
**Pull request by npadgen**
Spatterlight releases are now being done from GitHub, with three binary variants being distributed.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not [already refused].
- [ ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Comment by miccal** [comment]
@npadgen can you please confirm that the `Spatterlight_Mavericks.zip` version is for all macOS versions from Mavericks to Mojave?

File: Casks/spatterlight.rb [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
 
-  url "http://ccxvii.net/spatterlight/download/spatterlight-#{version}.zip"
-  appcast 'http://ccxvii.net/spatterlight/'
+  if MacOS.version >= :catalina
+    # URL_SECTION recent releases are only on github and not ccxvii.net
+    url "https://github.com/angstsmurf/spatterlight/releases/download/v#{version}/Spatterlight_Catalina.zip"
+    sha256 '4f9f975bbdc747fe6ad0a33ed25ff1baf82340987b2f6937f8a08ff83c337b79'
+
+  elsif MacOS.version >= :mavericks
+    # URL_SECTION recent releases are only on github and not ccxvii.net
```
Comment (vitorgalvao) [inline Casks/spatterlight.rb:10]: These need to be of the form `URL_SECTION was verified as official when first introduced to the cask`.

File: Casks/spatterlight.rb [hunk 2]
```diff
@@ -1,9 +1,23 @@
 cask 'spatterlight' do
-  version '0.5.0'
-  sha256 '1729c51676f791149f4829454318f373eff43bf8d8388ecbe3b345308c669ba1'
+  version '0.5.12b'
 
-  url "http://ccxvii.net/spatterlight/download/spatterlight-#{version}.zip"
-  appcast 'http://ccxvii.net/spatterlight/'
+  if MacOS.version >= :catalina
```
Comment (vitorgalvao) [inline Casks/spatterlight.rb:4]: This is backwards. Conditionals need to be `<=`.

File: Casks/spatterlight.rb [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+    # URL_SECTION recent releases are only on github and not ccxvii.net
+    url "https://github.com/angstsmurf/spatterlight/releases/download/v#{version}/Spatterlight_Catalina.zip"
+    sha256 '4f9f975bbdc747fe6ad0a33ed25ff1baf82340987b2f6937f8a08ff83c337b79'
+
```
Comment (vitorgalvao) [inline Casks/spatterlight.rb:8]: Extraneous empty line.

**Comment by vitorgalvao** [comment]
This looks to be a beta. This repo is for stable versions.

**Comment by npadgen** [comment]
Apologies, I have resubmitted the PR to homebrew/homebrew-cask-versions.

**Comment by vitorgalvao** [comment]
> Apologies, I have resubmitted the PR to homebrew/homebrew-cask-versions.

Don’t. The stable version isn’t nearly popular enough to justify adding the beta.

```
spatterlight
30 days: 1 (#5439)
90 days: 3 (#5481)
365 days: 22 (#5112)
Age: 1501 days (added 2016, April 01)
```

File: Casks/spatterlight.rb [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
+  elsif MacOS.version >= :mavericks
+    # URL_SECTION recent releases are only on github and not ccxvii.net
+    url "https://github.com/angstsmurf/spatterlight/releases/download/v#{version}/Spatterlight_Mavericks.zip"
+    sha256 'f9fa79d5d8e81397058e0f4ccd1d4e944ef6c5c66b2dc92792456feeaf8379e6'
+
+  else
+    # URL_SECTION recent releases are only on github and not ccxvii.net
+    url "https://github.com/angstsmurf/spatterlight/releases/download/v#{version}/Spatterlight_Lion.zip"
```
Comment (vitorgalvao) [inline Casks/spatterlight.rb:16]: Lion is too old. We don’t support that.

**Pull request closed without merging**